src/hotspot/share/oops/methodFlags.hpp

```diff
@@ -86,8 +87,8 @@ class MethodFlags {
 #undef M_STATUS_GET_SET
 
   int as_int() const { return _status; }
-  void atomic_set_bits(u4 bits);
-  void atomic_clear_bits(u4 bits);
+  void atomic_set_bits(u4 bits)   { Atomic::fetch_then_or(&_status, bits); }
+  void atomic_clear_bits(u4 bits) { u4 val = (~bits); Atomic::fetch_then_and(&_status, val); }
```

Comment:
Why introduce a new variable (and why the extra parens).  Just
```
Atomic::fetch_then_and(&_status, ~bits);
```

Reply:
The template didn't like this, I suppose I could add some casting.